### ISSUE TITLE
Undetected unused parameter

### ISSUE THREAD

**Issue by GaryJones** [sirbrillig/phpcs-variable-analysis#4]
In v1.0, the following detected `$context` as an unused variable (warning):

```php
// deep in an array
'is_needed' => function ( $context ) {
	return true;
},
```

Since v2.0, this doesn't seem to be seen. Likewise, the following unused `$foo` is also not reported:

```php
class Foo {
	/**
	 * Bar.
	 */
	public function bar( $foo ) {
		echo 'Foo::bar()';
	}
//...
}
```
Even a simple function doesn't report it:

```php
function test($foo) {

}
```

The following does report `$bar` as unused, so I'm confident that the VariableAnalysis is installed correctly and registered:

```php
function test($foo) {
	$bar = '';
}
```
```txt
37 | WARNING | [ ] Unused variable $bar. (VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable)
```

The relevant bit of my `.phpcs.xml.dist`:
```xml
<rule ref="VariableAnalysis"/>
```

**Comment by sirbrillig** [comment]
🤔  reproduced. Thanks for noticing this!

This is very interesting since the 2.0 change didn't modify the sniffs. I'll do more testing and get back to you (mostly afk this week).

**Comment by sirbrillig** [comment]
I've tracked it down to [this line](https://github.com/sirbrillig/phpcs-variable-analysis/blob/master/VariableAnalysis/Sniffs/CodeAnalysis/VariableAnalysisSniff.php?utf8=%E2%9C%93#L1466) where `allowUnusedFunctionParameters` is set to true.

It seems to be because that's [set in the example](https://github.com/sirbrillig/phpcs-variable-analysis/blob/master/VariableAnalysis/ruleset.xml#L50) in ruleset.xml. I'll strip it out of there.
